python/lsst/ts/externalscripts/base_take_ptc_flats.py

```diff
@@ -0,0 +1,451 @@
+                    type: string
+                    enum: ["CURRENT", "CHARGE"]
+                    default: "CURRENT"
+                  range:
+                    description:  >
+                      Electrometer measurement range. -1 for autorange.
+                    type: number
+                    default: -1
+                  integration_time:
+                    description: Electrometer integration time.
+                    type: number
+                    default: 0.1
+              ignore:
+                description: >-
+                    CSCs from the camera group to ignore in status check.
+                    Name must match those in self.group.components.
+                type: array
+                items:
+                  type: string
+
+            additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super().get_schema()
+
+        for properties in base_schema_dict["properties"]:
+            schema_dict["properties"][properties] = base_schema_dict["properties"][
+                properties
+            ]
+
+        return schema_dict
+
+    async def configure(self, config: types.SimpleNamespace):
+        """Configure script components including camera and electrometer.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        await self.configure_camera()
+
+        if hasattr(config, "electrometer_scan"):
+            await self.configure_electrometer(config.electrometer_scan["index"])
+
+        if hasattr(config, "ignore"):
+            for comp in config.ignore:
+                if comp in self.camera.components_attr:
+                    self.log.debug(f"Ignoring Camera component {comp}.")
+                    setattr(self.camera.check, comp, False)
+                else:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Group. "
+                        f"Must be one of {self.camera.components_attr}. "
+                        f"Ignoring."
+                    )
+
+        # Ensure flats_exp_times is provided in pairs
+        if len(config.flats_exp_times) % 2 != 0:
+            raise ValueError("Flats exposure times must be provided in pairs.")
+
+        # Check that each pair in flats_exp_times contains equal values
+        for i in range(0, len(config.flats_exp_times), 2):
+            if config.flats_exp_times[i] != config.flats_exp_times[i + 1]:
+                raise ValueError(
+                    "Flats exposure times must be provided in pairs of equal values."
+                )
+
+        # Handle interleave darks settings
+        if hasattr(config, "interleave_darks"):
+            if (
+                isinstance(config.interleave_darks["dark_exp_times"], list)
+                and config.interleave_darks["n_darks"]
+            ):
+                raise ValueError(
```

Comment:
Should be `RuntimeError`. Also, why not just ignore this altogether? just log a warning saying that `n_darks` are going to be ignored because `dark_exp_times` is an array.